crosspoint_reader/driver.py

```
@@ -293,47 +330,66 @@ def upload_books(self, files, names, on_card=None, end_session=True, metadata=No
-                chunk_size=chunk_size,
-                debug=debug,
-                progress_cb=_progress,
-                logger=self._log,
-            )
-            paths.append((lpath, os.path.getsize(filepath)))
-
-        self.report_progress(1.0, 'Transferring books to device...')
+        temp_files = []  # Track temp files for cleanup
+        
+        try:
+            for i, (infile, name) in enumerate(zip(files, names)):
```

Ruff (0.15.2)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)